components/AmountInputCard.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import { Grid, Card, Input, Link } from '@makerdao/ui-components-core';
+import { TextBlock } from './Typography';
+
+export default function({
+  max,
+  unit,
+  title,
+  update,
+  setValid,
+  getErrorMessage,
+  children
+}) {
+  const [error, setError] = useState();
+  const [value, setValue] = useState('');
+
+  const setMax = () => {
+    setValue(max.toBigNumber().toString());
+    update(max);
+  };
+
+  const onChange = event => {
+    const { value } = event.target;
+    setValue(value);
+
+    if (isNaN(parseFloat(value))) {
+      setError('Please enter a valid number');
+      setValid(false);
+      return;
+    }
+
+    let newError;
+    try {
+      const typedValue = unit(value);
+      newError = getErrorMessage(typedValue);
+      if (!newError) update(typedValue);
+    } catch (e) {
```

Comment:
we use a try/catch block in case the Currency instantiation fails for some reason (e.g. a bad value for the `unit` prop was not passed in)